// math-formulas/index.js
function squareArea(side) {
    return side * side;
}

function rectangleArea(length, bredth) {
    return length * bredth;
}

function triangleArea(base, height) {
    return (base * height) / 2;
}

function circleArea(radius) {
    return Math.PI * radius * radius;
}

function cubeSurfaceArea(side) {
    return 4 * side * side;
}

function cubeTotalSurfaceArea(side) {
    return 6 * side * side;
}

function cuboidSurfaceArea(length, bredth, height) {
    return 2 * height * (length + bredth);
}

function cuboidTotalSurfaceArea(length, bredth, height) {
    return 2 * (length * bredth + bredth * height + height * length);
}

function cylinderSurfaceArea(radius, height) {
    return 2 * Math.PI * radius * height;
}

function cylinderTotalSurfaceArea(radius, height) {
    return 2 * Math.PI * radius * (radius + height);
}

function coneSurfaceArea(radius, slantHeight) {
    return Math.PI * radius * slantHeight;
}

function coneTotalSurfaceArea(radius, slantHeight) {
    return Math.PI * radius * (radius + slantHeight);
}

function sphereSurfaceArea(radius) {
    return 4 * Math.PI * radius * radius;
}

function sphereTotalSurfaceArea(radius) {
    return 4 * Math.PI * radius * radius;
}

function hemisphereSurfaceArea(radius) {
    return 2 * Math.PI * radius * radius;
}

function hemisphereTotalSurfaceArea(radius) {
    return 3 * Math.PI * radius * radius;
}


module.exports = {
    squareArea, 
    rectangleArea, 
    triangleArea, 
    circleArea, 
    cubeSurfaceArea, 
    cubeTotalSurfaceArea, 
    cuboidSurfaceArea,
    cuboidTotalSurfaceArea,
    cylinderSurfaceArea,
    cylinderTotalSurfaceArea,
    coneSurfaceArea,
    coneTotalSurfaceArea,
    sphereSurfaceArea,
    sphereTotalSurfaceArea,
    hemisphereSurfaceArea,
    hemisphereTotalSurfaceArea
};
